controllers/tab/pubIds/form/PublicIdentifiersForm.inc.php

```diff
@@ -0,0 +1,174 @@
+<?php
+
+/**
+ * @file controllers/tab/issueEntry/form/PublicIdentifiersForm.inc.php
+ *
+ * Copyright (c) 2014-2015 Simon Fraser University Library
+ * Copyright (c) 2003-2015 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class PublicIdentifiersForm
+ * @ingroup controllers_tab_issueEntry_form_PublicIdentifiersForm
+ *
+ * @brief Displays a submission's pub ids form.
+ */
+
+import('lib.pkp.classes.form.Form');
+
+class PublicIdentifiersForm extends Form {
+
+	/** @var int The context id */
+	var $_contextId;
+
+	/** @var object The pub object the identifiers are edited of 
+	 * (Submission, Issue)
+	 */
+	var $_pubObject;
+
+	/** @var int The current stage id */
+	var $_stageId;
+
+	/**
+	 * @var array Parameters to configure the form template.
+	 */
+	var $_formParams;
+
+	/**
+	 * Constructor.
+	 * @param $pubObject object
+	 * @param $stageId integer
+	 * @param $formParams array
+	 */
+	function PublicIdentifiersForm($pubObject, $stageId = null, $formParams = null) {
+		parent::Form('controllers/tab/pubIds/form/publicIdentifiersForm.tpl');
+
+		$this->_pubObject = $pubObject;
+		$this->_stageId = $stageId;
+		$this->_formParams = $formParams;
+
+		$request = Application::getRequest();
+		$context = $request->getContext();
+		$this->_contextId = $context->getId();
+
+		$this->addCheck(new FormValidatorPost($this));
+	}
+
+	/**
+	 * Fetch the HTML contents of the form.
+	 * @param $request PKPRequest
+	 * return string
+	 */
+	function fetch($request) {
+		$templateMgr = TemplateManager::getManager($request);
+		$pubIdPlugins =& PluginRegistry::loadCategory('pubIds', true, $this->getContextId());
+		$templateMgr->assign('pubIdPlugins', $pubIdPlugins);
+		$templateMgr->assign('pubObject', $this->getPubObject());
+		$templateMgr->assign('stageId', $this->getStageId());
+		$templateMgr->assign('formParams', $this->getFormParams());
+		return parent::fetch($request);
+	}
+
+	/**
+	 * Initialize form data.
+	 */
+	function initData() {
+		$pubObject = $this->getPubObject();
+		import('lib.pkp.classes.plugins.PKPPubIdPluginHelper');
+		$pubIdPluginHelper = new PKPPubIdPluginHelper();
+		$pubIdPluginHelper->init($this->getContextId(), $this, $pubObject);
+	}
+
+
+	//
+	// Getters and Setters
+	//
+	/**
+	 * Get the pub object
+	 * @return object
+	 */
+	function getPubObject() {
+		return $this->_pubObject;
+	}
+
+	/**
+	 * Get the stage id
+	 * @return int
```

Comment:
(Wherever possible, describe the namespace of defined numbers -- e.g.:

    @return int WORKFLOW_STAGE_ID_...